examples/inmemory_storage.py

```diff
@@ -0,0 +1,41 @@
+import pprint
+
+from asab.storage.mongodb import StorageService
+
+import asab
+import asab.storage
+
+
+'''
+../etc/site.conf
+
+[asab:storage]
+type=inmemory
+
+'''
+class MyApplication(asab.Application):
+
+	async def initialize(self):
+		# Loading the web service module
+		self.add_module(asab.storage.Module)
+
+
+	async def main(self):
+		storage = self.get_service("asab.StorageService")
+
+		u = storage.upsertor("test-collection", 1, version=0)
```

Comment:
This is not correct.
You have to specify a object id and keep a version as None ...

```
objid = 1
u = storage.upsertor("test-collection", obj_id =objid)
```

You should specify a version only if you retrieved that from an existing object, it is kind of a lock mechanism.

There are basically three scenarios for `obj_id`:
1) You are updating an existing object so you have a `obj_id` and a `version` from existing instance of the object
2) You create a new object and you want to let the engine to generate `obj_id` for you ... then you should not specify `obj_id` and `version`
3) You create a new object and you want to specify your own `obj_id` ... then you provide the `obj_id` and don't provide `version`

^^^ - add this to a documentation of upsertor method and adjust the code accordingly.

Reply:
Not specifying `version` throws KeyError

```
Traceback (most recent call last):
  File "/asab/asab/application.py", line 283, in run
    task.result()
  File "./inmemory_storage.py", line 28, in main
    objid = await u.execute()
  File "/asab/asab/storage/inmemory.py", line 19, in execute
    obj = await self.Storage.get(self.Collection, self.ObjId)
  File "/asab/asab/storage/inmemory.py", line 72, in get
    coll = self.InMemoryCollections[collection]
KeyError: 'test-collection'
```

Updating `get` method with `try/except` on `inmemory` storage could fix it

```py
async def get(self, collection:str, obj_id):
	try:
		coll = self.InMemoryCollections[collection]
		return coll[obj_id]
	except KeyError:
		return None

```

What's your thoughts?

Reply:
Then it has to be fixed in InMemory implementation ... that should not be difficult, it is just about a default value, which is 1.

Reply:
The KeyError is happening on `self.InMemoryCollections[collection]` where `self.InMemoryCollections` is empty dict. The call goes from `execute` where it calls `Storage.get` (`"/asab/asab/storage/inmemory.py", line 19, in execute`) before setting the value..

Reply:
And don't do that try/catch ... that's hacking, not software engineering ;-)

Reply:
The better approach is to do try / except KeyError on `File "/asab/asab/storage/inmemory.py", line 19, in execute` ... and handle an exception properly.

Reply:
Defaulting `version` to zero resolves the `KeyError`. I added your comment as docstring to `StorageServiceABC.upsertor`